spec/lang/step/terminators.md

```diff
@@ -174,6 +174,25 @@ impl<M: Memory> Machine<M> {
         })
     }
 
+    /// A helper function to evaluate the return expression and prepare it for in-place passing.
+    fn eval_return_place(&mut self, ret_expr: PlaceExpr) -> NdResult<(Place<M>, Type)> {
+        // First evaluate the return place. (Left-to-right!)
+        let (caller_ret_place, caller_ret_ty) = self.eval_place(ret_expr)?;
+        // FIXME: should we care about `caller_ret_place.align`?
+        // Make sure we can use it in-place.
+        self.prepare_for_inplace_passing(caller_ret_place, caller_ret_ty)?;
+        ret((caller_ret_place, caller_ret_ty))
+    }
+
+    /// A helper function to evaluate a function pointer.
+    fn eval_function_ptr(&mut self, fn_ptr: ValueExpr) -> Result<Function> {
+        // Then evaluate the function that will be called.
```

Comment:
This comment clearly was part of a sequence of comments and you moved it around, breaking the sequence and thus ripping this comment out of its context.

Please maintain the original sequence of comments in `eval_terminator`.